modules/rostests/rosautotest/main.cpp

```diff
@@ -139,6 +151,23 @@ wmain(int argc, wchar_t* argv[])
             WineTest.Run();
         }
 
+        /* Clear the stringstream */
+        ss.str("");
+        ss.clear();
+
+        /* Show the beginning time again */
+        ss << "[ROSAUTOTEST] System uptime at start was " << setprecision(2) << fixed;
+        ss << ((float)TestStartTime / 1000) << " seconds" << endl;
+
+        /* Show the time now so that we can see how long the tests took */
+        TestEndTime = GetTickCount();
+        ss << endl
+           << "[ROSAUTOTEST] System uptime at end was " << setprecision(2) << fixed;
+        ss << ((float)TestEndTime / 1000) << " seconds" << endl;
+        ss << "[ROSAUTOTEST] Duration was " << (((float)TestEndTime - (float)TestStartTime) / 1000) / 60;
+        ss << " minutes" << endl;
```

Comment:
Can't this output 1 line only, like ' duration (end - start)', instead of 4?

Reply:
@SergeGautherie, It could be, but I prefer what is done here. A typical testbot run contains over 75,000 lines of output and saving three lines is very trivial in this context. Thanks for your suggestions though. They are always welcome.